### ISSUE TITLE
KafkaAdminClient create object error

### ISSUE THREAD

**Issue by hatcherfang** [dpkp/kafka-python#1800]
description:
use package KafkaAdminClient create topics error.
traceback logs as below:
```
  File "/home/xx/venv/lib/python3.6/site-packages/kafka/admin/client.py", line 205, in __init__
    self._refresh_controller_id()
  File "/home/xx/venv/lib/python3.6/site-packages/kafka/admin/client.py", line 249, in _refresh_controller_id
    version = self._matching_api_version(MetadataRequest)
  File "/home/xx/venv/lib/python3.6/site-packages/kafka/admin/client.py", line 230, in _matching_api_version
    self._client.get_api_versions()[operation[0].API_KEY][1])
TypeError: 'NoneType' object is not subscriptable
```

reproduce code:
```
from kafka import KafkaAdminClient
config = {'bootstrap_servers': 'xx.xx.xx.xx:9092', 'api_version': (1, 0, 1)}
adminCli = KafkaAdminClient(**config)
```

**Comment by hatcherfang** [comment]
I solved it by removing the 'api_version' and make sure your network can connect the brokers successfully.

code as below:
```
from kafka import KafkaAdminClient
config = {'bootstrap_servers': 'xx.xx.xx.xx:9092'}
adminCli = KafkaAdminClient(**config)
```

**Comment by michalspondr** [comment]
Without _api_version_ parameter I get:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cdp/base/venv/lib/python3.7/site-packages/kafka/admin/client.py", line 198, in __init__
    **self.config)
  File "/home/cdp/base/venv/lib/python3.7/site-packages/kafka/client_async.py", line 239, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/cdp/base/venv/lib/python3.7/site-packages/kafka/client_async.py", line 865, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
```

**Comment by michalspondr** [comment]
It seems this is a duplicate of #1675.

**Comment by ian28223** [comment]
`TypeError: 'NoneType' object is not subscriptable`  is fixed with 1.4.7 (https://github.com/dpkp/kafka-python/commit/98ebff87a78bafbb15dd95c5174c5a1041a848ed)

but would now raise `IncompatibleBrokerVersion` since `self._client.get_api_versions()` always returns None. My PR to fix that here https://github.com/dpkp/kafka-python/pull/1953

**Comment by ccmanito** [comment]
description:
use package KafkaAdminClient  error.
traceback logs as below:
`# kafka-python
# 获取topic的配置信息
def get_topic_config():
    resource_config = ConfigResource(ConfigResourceType.TOPIC, "test_midop")
    config_entries = admin.describe_configs([resource_config])
    print(config_entries.resources)

    # ERROR:
        # kafka.errors.IncompatibleBrokerVersion: IncompatibleBrokerVersion: Kafka broker does not support the 'DescribeConfig
        # sRequest_v0' Kafka protocol.

if __name__ == "__main__":
    admin = KafkaAdminClient(bootstrap_servers='10.226.132.131:9092', api_version=(0,10,2))
    get_topic_config()`

reproduce code:
`Traceback (most recent call last):
  File "describe.py", line 35, in <module>
    admin = KafkaAdminClient(bootstrap_servers='10.226.132.131:9092', api_version=(0,1
0,2))
  File "/Users/chengshiwei/worker/cmw-kafka/.env/lib/python3.7/site-packages/kafka/adm
in/client.py", line 211, in __init__
    self._refresh_controller_id()
  File "/Users/chengshiwei/worker/cmw-kafka/.env/lib/python3.7/site-packages/kafka/adm
in/client.py", line 261, in _refresh_controller_id
    version = self._matching_api_version(MetadataRequest)
  File "/Users/chengshiwei/worker/cmw-kafka/.env/lib/python3.7/site-packages/kafka/adm
in/client.py", line 240, in _matching_api_version
    .format(operation[0].__name__))
kafka.errors.IncompatibleBrokerVersion: IncompatibleBrokerVersion: Kafka broker does n
ot support the 'MetadataRequest_v0' Kafka protocol.`

**Comment by ccmanito** [comment]
use  describe_configs  eeror:
description:
`def get_topic_config():
    resource_config = ConfigResource(ConfigResourceType.TOPIC, "test_midop")
    config_entries = admin.describe_configs([resource_config])
    print(config_entries.resources)
if __name__ == "__main__":
    admin = KafkaAdminClient(bootstrap_servers='localhost:9092', api_version=(0,10,2))
    get_topic_config()`
reproduce code: 
`kafka.errors.IncompatibleBrokerVersion: IncompatibleBrokerVersion: Kafka broker does n
ot support the 'MetadataRequest_v0' Kafka protocol.`

Is it because kafka version 0.10.2.1 is not supported？

**Comment by jeffwidman** [comment]
#1953 is merged and available in the `2.0.0` release, so closing.
